File: 0724-find-pivot-index/prefix-sum/python/solution.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        total = sum(nums)
        left_sum = 0
        for idx, num in enumerate(nums):
            right_sum = total - left_sum - num
            if left_sum == right_sum:
                return idx
            left_sum += num
        return -1

File: 0724-find-pivot-index/prefix-sum/cpp/solution.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int total = std::accumulate(nums.begin(), nums.end(), 0);
        int left_sum = 0;

        for (int idx = 0; idx < nums.size(); ++idx) {
            int num = nums[idx];
            int right_sum = total - left_sum - num;
            if (left_sum == right_sum) {
                return idx;
            }
            left_sum += num;
        }

        return -1;
    }
};
